cmd/infra/powervs/destroy.go

```diff
@@ -371,8 +376,78 @@ func deleteSecrets(name, namespace, cloudInstanceID string, accountID string, re
 	return nil
 }
 
+// destroyPowerVsDhcpServer destroying powervs dhcp server
+func destroyPowerVsDhcpServer(ctx context.Context, logger logr.Logger, cloudInstanceID string, options *DestroyInfraOptions, session *ibmpisession.IBMPISession) error {
+	// Attempt dhcp server deletion only when cloud instance is created by hypershift
+	// Do not clean up, if PowerVS instance is user provided
+	if options.CloudInstanceID == "" {
```

Comment:
```suggestion
	if options.CloudInstanceID != "" { 
	    return nil
	}
	client := instance.NewIBMPIDhcpClient(ctx, session, cloudInstanceID)
```
We can do an early return to keep the code more readable IMO.

Reply:
added it, thanks!